graphql_compiler/global_utils.py

```diff
@@ -9,6 +9,13 @@
 from .ast_manipulation import safe_parse_graphql
 
 
+# Imported from here, to avoid spreading the conditional import everywhere.
+try:
+    from functools import cached_property  # noqa
+except ImportError:
+    from backports.cached_property import cached_property  # type: ignore[no-redef] # noqa
```

Comment:
See https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker if you want to learn more about `# type: ignore[no-redef]` — this is something we should get used to using everywhere since it disables only a particular type of mypy check.